### ISSUE TITLE
Training a New Language Model: Loss Oscillation and Non-Convergence

### ISSUE THREAD

**Issue by damon-93** [SWivid/F5-TTS#838]
### Checks

- [x] This template is only for usage issues encountered.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

Python 3.10.0
torch  2.4.1+cu124

### Training a New Language Model: Loss Oscillation and Non-Convergence

Hello, this is a great project! I am trying to train an Indonesian language model, but I am encountering an issue where the loss oscillates and does not converge.

### Dataset Processing:
I am using the validated.tsv wav files from the cv-corpus-20.0-2024-12-06 dataset. To ensure that every step is processed correctly, I did not use the text labels from validated.tsv. Instead, I followed the process in WeUI:
* 1. Transcribe Data
* 2.Voca Check → Extend
* 3.Prepare Data

After processing, I obtained 34,074 wav files for training, with approximately 31 hours of audio.

### Training Configuration:
I used Auto Settings and did not select FineTune. My training configuration is as follows:
```
{
    "exp_name": "F5TTS_Base",
    "learning_rate": 7.5e-05,
    "batch_size_per_gpu": 243,
    "batch_size_type": "frame",
    "max_samples": 64,
    "grad_accumulation_steps": 1,
    "max_grad_norm": 1,
    "epochs": 88,
    "num_warmup_updates": 1704,
    "save_per_updates": 3408,
    "keep_last_n_checkpoints": 5,
    "last_per_updates": 852,
    "finetune": false,
    "file_checkpoint_train": "",
    "tokenizer_type": "char",
    "tokenizer_file": "",
    "mixed_precision": "fp16",
    "logger": "tensorboard",
    "bnb_optimizer": false
}
```

### ❌ Actual Behavior

Currently, my loss is oscillating and not converging. Could you provide me with some guidance on how to resolve this issue? What configuration parameters should I modify, or how should I approach this problem?

![Image](https://github.com/user-attachments/assets/3a4397a9-db6f-4bd3-816b-d41edc63f2d5)

**Comment by damon-93** [comment]
I used noisereduce to filter the noise, and also tried to increase epochs, but all failed to solve the problem. I have no idea, and I need some help or advice~

**Comment by SWivid** [comment]
if train from scratch with multi-gpu, i would recommand you to use train.py

if single gpu and want to keep with finetune gradio:
{
    "exp_name": "F5TTS_Base",
    "learning_rate": 7.5e-05,
    "batch_size_per_gpu": 3600,
    "batch_size_type": "frame",
    "max_samples": 64,
    "grad_accumulation_steps": 1,
    "max_grad_norm": 1,
    "epochs": 100,
    "num_warmup_updates": 5000,
    "save_per_updates": 100000,
    "keep_last_n_checkpoints": 5,
    "last_per_updates": 10000,
    "finetune": false,
    "file_checkpoint_train": "",
    "tokenizer_type": "char",
    "tokenizer_file": "",
    "mixed_precision": "fp16",
    "logger": "tensorboard",
    "bnb_optimizer": false
}

gradio method for training/finetuning needs updates

**Comment by SWivid** [comment]
for loss, you could 

> I have searched for existing issues, including closed ones, and couldn't find a solution.

**Comment by SachinTelecmi** [comment]
![Image](https://github.com/user-attachments/assets/79701e51-45e6-4d0f-ab00-90bb98688563)

![Image](https://github.com/user-attachments/assets/4e63484f-dd37-48f8-90cf-03f8892d0489)
@SWivid can you tell me seeing the loss curve what more i can do right now model is still struggling to generate audio properly altho it is able to utter some words properly but most of the words fail 
any suggestion is highly appreciated very high oscillation in the loss curve

**Comment by awei669** [comment]
@damon-93 The loss seems to fluctuate because the condition is dropped with some probability. If I turn off the condition dropout, the loss becomes much more stable.

**Comment by rasenganai** [comment]
@damon-93 have you tried increasing the batchsize?
243 is like 2s per batch
